core/query/src/main/java/datawave/core/query/logic/QueryKey.java

```diff
@@ -0,0 +1,104 @@
+package datawave.core.query.logic;
+
+import java.io.Serializable;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class QueryKey implements Serializable {
+    private static final long serialVersionUID = -2589618312956104322L;
```

Comment:
I have never understood how these are generated, or even what their usefulness is.  @ivakegg?

Reply:
the serialVersionUID has to be assigned for any Serializable class for compatibility. This value is likely the value automatically generated by the compiler, but it can also be versioned manually for compatibility

Reply:
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/Serializable.html